spring-integration-file/src/main/java/org/springframework/integration/file/FileWritingMessageHandler.java

```diff
@@ -645,6 +663,10 @@ private File handleInputStreamMessage(final InputStream sourceFileInputStream, F
 
 				@Override
 				protected void whileLocked() throws IOException {
+					if (FileWritingMessageHandler.this.newFileCallback != null && !fileToWriteTo.exists()) {
+						FileWritingMessageHandler.this.newFileCallback.accept(fileToWriteTo, requestMessage);
```

Comment:
Since this is called for a raw `InputStream` payload we need `&& append` here.

OTOH, I wonder why we only want to restrict this to be only called when appending. Perhaps we should always call it when a new file is about to be created. It would need some more logic in `handleFileMessage`. i.e. only rename the input file if we are not in an append mode, and want to delete the inbound file, and we **don't** have a `newFileCallback`.

Reply:
Oops; yes; you are correct; sorry for the noise.

Ok, @artembilan let's leave as-is (append mode only).